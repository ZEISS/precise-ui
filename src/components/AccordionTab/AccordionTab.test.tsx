import * as React from 'react';
import * as enzyme from 'enzyme';
import { AccordionTab } from './';

describe('<AccordionTab />', () => {
  it('should render an empty <AccordionTab> component', () => {
    const wrapper = enzyme.shallow(<AccordionTab header="Header" />);
    expect(wrapper.text()).toBe('');
  });

  it('should render a text as a content', () => {
    const wrapper = enzyme.shallow(<AccordionTab header="Header">Content</AccordionTab>);
    expect(wrapper.text()).toBe('Content');
  });

  it('should render a component as a content', () => {
    const wrapper = enzyme.shallow(
      <AccordionTab header="Header">
        <div>Content</div>
      </AccordionTab>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a header as a component', () => {
    const wrapper = enzyme.mount(<AccordionTab header={<div>Header</div>}>Content of the tab</AccordionTab>);
    expect(wrapper).toMatchSnapshot();
  });
});
