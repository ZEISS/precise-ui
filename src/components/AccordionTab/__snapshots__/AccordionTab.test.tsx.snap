// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionTab /> should render a component as a content 1`] = `
<div>
  <div>
    Content
  </div>
</div>
`;

exports[`<AccordionTab /> should render a header as a component 1`] = `
<AccordionTab
  header={
    <div>
      Header
    </div>
  }
>
  <div>
    Content of the tab
  </div>
</AccordionTab>
`;
